lib/ansible/modules/remote_management/manageiq/manageiq_user.py

```diff
@@ -183,7 +190,7 @@ def compare_user(self, user, name, group_id, password, email):
         """
         found_difference = (
             (name and user['name'] != name) or
-            (password is not None) or
+            (password is not None and self.module.params['update_password'] == 'always') or
```

Comment:
> The desired behavior was that I can have a playbook that lays down MIQ users. Later, the users change their passwords. I re-run the 'ensure-users-exists' playbook, and it doesn't lay down the old/original passwords during the re-run.

Great ! , then we need to **delete the passwrod** if a user exist, like @cben pointed out, current approach may re-set the password inadvertently  if user changes,
so we do not need the check here, all we need is to delete the password if the user exits and we are "on_create", this check will fail because we will not have password in the params

Reply:
Nice @cben 👍 ( I didn't understand what @joelddiaz intended here ... )